app/controllers/websites_controller.rb

```diff
@@ -6,7 +6,9 @@ def index
   end
 
   def update
-    @website.update_attributes(website_params)
+    strong_params = website_params
+    strong_params[:origins] = current_website.ensure_valid_protocol_for_origins(website_params[:origins])
+    @website.update_attributes(strong_params)
```

Comment:
Here we need to use some if clause to make sure that we check for validations. That way we can render the :edit action again if there is an error and show that error using simple_forms in the form.

Also that ensure_valid_protocol_for_origins should be as a callback in website to make sure that we always call it not here in the controller.

Use `before_validation :check_origins, if: :origins_changed?` something like this.